### ISSUE TITLE
Docs: [no-inferrable-types] Explain whether and how to use (or not use) with `isolatedDeclarations`

### ISSUE THREAD

**Issue by pikadun** [typescript-eslint/typescript-eslint#9125]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-inferrable-types/

### Description

It would be nice to use "no-inferrable-types" with [isolatedDeclarations](https://github.com/microsoft/TypeScript/issues/57475)

### Fail

```TypeScript
const a: bigint = 10n;
export const b = true;
```

### Pass

```TypeScript
const a = 10n;
export const b: boolean = true;
```

**Comment by Josh-Cena** [comment]
Strong -1 on encoding TS semantics in our rule. ID exists almost precisely to eliminate "inferred types" so I'm not sure why anyone wants to use it with `no-inferrable-types`. We can add to docs saying they are incompatible.
